### ISSUE TITLE
Your setup? / inspiration

### ISSUE THREAD

**Issue by evlo** [dinvlad/HyperVision#1]
I'm thinking about doing something like ambilight for my tv, looked at the commercial solutions and they look kinda ugly. So I was wondering if you could share you setup and what camera you use.

I was thinking maybe putting the camera into light fixture - but i guess wireless would not be fast enough for image transmission and I don't even know how it would work as camera cam for hypervision or just maybe tiny little fish eye camera just pockint through small hole of the tv cabinet. 

Basically I'm looking for inspiration how to hide the cam.

**Comment by dinvlad** [comment]
Thanks for reaching out @evlo - tbh I've never finished wrapping it up with making the camera to look nice. I was considering 3D-printing a fixture very similar to [AmbiVision](https://ambivision.tv/) though, which looks nice to me. The camera setup when I tested it, was directly below my TV, and it was [this](https://smile.amazon.com/gp/product/B013JWEGJQ) fish-eye RPi camera with 194° FoV. I was getting decent results with it, but the main problem was in color reproduction at the corners/borders of the TV, so unfortunately this solution was not ideal still. I haven't explored this space further since, and been using [this LED strip](https://www.superlightingled.com/360-degree-led-neon-sign-light-waterproof-ip68-silicone-housing-p-2929.html) in static color mode, which looks slick and very similar to the (much more expensive) [Philips Hue Play Gradient Lightstrip](https://www.philips-hue.com/en-us/p/hue-white-and-color-ambiance-play-gradient-lightstrip-75-inch/046677560423).

Btw your setup looks really clean - if you already have other components (LED strip + power, Raspberry Pi 3B+/4), then get an inexpensive camera like that one and test if you like the color matching. Then if you like it, it's probably worth 3D-printing a shell like that. I also got [this](https://smile.amazon.com/gp/product/B07J68TJ7L) Pi cam extension cable, which would look clean/stealthy on my black TV stand, but in your case you might want the white version, probably like [this](https://smile.amazon.com/Arducam-Octoprint-Monitor-Extension-Raspberry/dp/B087FDJ2RP) one.

All in all, I'd say it's not an inexpensive project, so do it only if you like tinkering 😄 Otherwise, maybe consider getting the commercial Ambivison, which looks quite slick and probably easier to get working properly (although I've heard it can be painful in some setups as well). I do think once 4k 120fps HDMI splitters hopefully become more widely available and inexpensive, it could be a superior experience compared to camera-based ones, mainly due to color matching. Although it might be possible to further tweak camera-based ones, but there're just not enough pixels to correctly approximate the color from, when it comes to the distorted view from the camera looking at TV directly from below, on top of these inexpensive cameras being of fairly poor quality color-capturing wise.

So if you have the budget (or maybe even as a budget-saving thing) it might be better to invest into a 1-in 2-out splitter like [this](https://www.amazon.com/HDMI-Splitter-1x2-4K-120Hz/dp/B0B7W94LHB), which would still work with Hyperion and be fully customizable to your needs thanks to it, but with superior color/latency and completely hidden from view.

**Comment by evlo** [comment]
Thank you for the long reply, to me the ambivision camera is quite unsightly anyways, so that is why I was looking at the alternatives. Also I use TV apps a lot so that is why i was looking into camera solution. 

I guess I will also stick with solid color for the time being. 

I don't really super care about color accuracy, I would play with the network attached camera, but I have no idea how to use it as source in the hyperion-ng. 

Currently I have just a simple led strip that can do single color, i do not have rPi connected to it etc.

**Comment by dinvlad** [comment]
Yeah I see - one cool alternative I'd recommend to try if you get an RGB/RGBW strip is matching its color temperature to the time of day, in addition to turning it on on the TV (if it's "smart" enough). This is something that would make literally a night and day difference in your experience with the TV - being more "awake" in the morning and gradually transition to a "warm"/calming color in the evening. I initially wanted to use my strip to match colors to movies and video games, but found it's kinda distracting anyway, and now I have it as just a cool background without all this fussiness of making it work and look good and not overly distracting/irritating with colors on the screen.

So you can easily add your LED strip to Hyperion, and then control Hyperion itself based on a script and/or using something like [Home Assistant](https://www.home-assistant.io/), see e.g. [here](https://www.home-assistant.io/integrations/hyperion/) and [here](https://community.home-assistant.io/t/adaptive-lighting-custom-component-that-automatically-adapts-the-brightness-and-color-of-your-lights-based-on-the-suns-temperature-but-stops-when-you-manually-make-a-change/257416/104). I have this setup, but was lazy and only set warm color in the evening 😄

**Comment by evlo** [comment]
I was kinda afraid it might be cool project but end up looking kinda tacky in the end. 
I went for no ambilight TV as samsung has much better, to my eyes, anti reflective coating so when you write that it was discrating i'm thinking maybe it is really no loss after all.

What you describe is pretty much my current setup, controlled with HA, only it goes from red to pink. It would make sense if i posted whole room photo.

So I'm think now i will go with quinled and wled and do like 30 minutes long super slow gradual transition between red and warm white in the eventing, but in sequence, just for the geek factor :)

**Comment by dinvlad** [comment]
Haha yeah, that sounds pretty sweet!

**Comment by robertzas** [comment]
Hi, I have a govee setup with the camera for my regular 75 inch TV, which seems to work well enough, but started looking for a similar setup for my future 120 projection screen plans. The HDMI capture strategy either seems to have some issues, namely, either an expensive splitter or lack of capability to run hdcp/4khdr content. This seems like the perfect solution for a projector since you can conveniently place the camera near the projector, and I am still in the process of finishing my basement(where the projector will be), so I can run some custom wiring across to the front to the screen anyway. 

Your YouTube results looks pretty perfect to me, but is there any reason this is worse then a HDMI capture strategy? Any noticable lag or anything else? It seems you mention potentially switching to an HDMI capture device, but why?

TYIA

**Comment by dinvlad** [comment]
I would suggest trying both, but as explained earlier HDMI provides just a superior quality and latency. With a camera mounted at the projector, you'd also need to think how you'd wire it into your TV LED strip (or if you send the signal over wifi/bluetooth, it'll decrease latency even further). And unlike a couple years ago when I built this, HDCP/4k HDR compatible frame grabbers are on the market now, for a comparable price. So in my mind the latter is now just better in all/most regards.

Also, I should add the video looks better than the reality in this case. I've noticed significant latency and poor color reproduction (I also believe youtube compression algos "hide" some of these issues experienced in-person).

**Comment by orichienal** [comment]
Hi folks, 
really a great extension.
I have two questions:
does this also work with the internal pi camera (raspberry pi zero 2 w)?
is there a possibility to do the whole thing with hyperbian and then via the terminal via ssh (hyperbian has no desktop)?

I would be very happy about feedback.
